Sources/Yams/Representer.swift

```diff
@@ -230,7 +228,8 @@ extension URL: ScalarRepresentable {
 extension String: ScalarRepresentable {
     /// This value's `Node.scalar` representation.
     public func represented() -> Node.Scalar {
-        return .init(self)
+        let scalar = Node.Scalar(self)
+        return scalar.resolvedTag.name == .str ? scalar : .init(self, Tag(.str), .singleQuoted)
```

Comment:
The performance impact can be almost ignored since `resolvedTag` was previously called in `serializeScalar(_:)` on serializing.
https://github.com/jpsim/Yams/blob/41c33b6f96/Sources/Yams/Emitter.swift#L413